components/layout/index.scss

```diff
@@ -0,0 +1,81 @@
+$sidebarWidth: 10rem;
+
+.layout {
+  grid-template-rows: min-content auto;
+  position: relative;
+  min-height: 100vh;
+  display: grid;
+  grid-column-gap: 1rem;
+  grid-template-columns: $sidebarWidth repeat(3, 1fr);
+  grid-template-areas: "header header header header"
+                       "sidebar main main main";
+
+  .layout-header {
+    grid-area: header;
+
+    .header-item:first-child {
+      flex: 0 0 $sidebarWidth;
+      margin-right: 1em;
+
+      .hamburger {
+        display: none;
+        z-index: 99;
+      }
+    }
+
+    .header-item:last-child {
+      flex: 0 0 $sidebarWidth;
+      margin-left: 1em;
+    }
+  }
+
+  .layout-sidebar {
+    height: 100%;
+    grid-area: sidebar;
+  }
+
+  .layout-content {
+    grid-area: main;
+  }
+}
+
+
+@media screen and (max-width: 992px) {
+  .header-item:first-child,
+  .header-item:last-child {
+    flex: 0 0 1rem !important;
+  }
+
+  .header-item:first-child {
+    > a {
+      display: none;
+    }
+
+    > .hamburger {
+      display: block !important;
+    }
+  }
+
+  .layout {
+    grid-template-columns: repeat(2, 1fr);
+    grid-template-areas: "header header"
+                         "main main";
+
+    > * {
+      padding-left: 1rem;
+      padding-right: 1rem;
+    }
+  }
+
+  .layout-sidebar {
+    position: absolute;
+    background: gray;
+    left: -$sidebarWidth;
```

Comment:
It's better to use `transform` because it has better performance and implicitly tells browser that the element should be painted in a separate layer because it might change.